### PR TITLE
meta: Revert "main" to "./lib/index", no dist (fix package on `master` branch)

### PR DESCRIPTION
**Pull request by DeeDeeG**
We don't have the `dist/` dir on `master` branch.

If package.json "main" field points to this non-existent folder, then the package is effectively broken.

This commit restores "main" to point to "./lib/index", as this seems to be working now in my tests.

We can pre-compile the `dist/` folder again when tagging the next release, if we really want to, but I'm pretty sure Pulsar compiles the code on first run, so the stakes for doing so seem pretty small to me. (Impact should be limited to improvements in first startup speed, if I understand correctly.)

\[ EDIT: Transpilation on-demand only works when the package is installed to `~/.pulsar/packages/`. It does **not** get auto-transpiled if it's a bundled/core package. Ah, well. We still have to transpile, I guess. \]

---

Note: according to Mauricio, the command to build the `dist/` dir is: `ELECTRON_VERSION=12 babel lib -d dist --compact auto --source-type module`

But it is not trivially obvious to me how to automate this at the moment, so I won't try to do so for now. This just gets the repo working on `master` branch again, without bothering with pre-compilation stuff.

\[ EDIT TO ADD, NOTE: This section below the divider line was documenting a previously undocumented build step, but was ultimately off-topic in terms of actionable items for this PR. Was strictly advisory, wanted to write it down so I did not forget how to do the step!!! This section was not actually germane to how to evaluate this PR on its own merits, IMO!!!!! Arguably I should not have mentioned it, since it is off-topic!!! Please don't let this bit distract you, reviewers! Thank you! \]

### CODE REVIEW

**Comment by DeeDeeG** [comment]
Tangential note: I hope to see the way the tests run here be improved, since as far as I can tell a lot of the tests are actually failing, and it is running tests for _the editor's own specs_, not the package specs? Or something along those lines?

Until then, I don't really think the test results apply or say much about the quality of this (or any) PR.

Thanks to the efforts to get those set up, they are as good a starting point as any.

**Comment by confused-Techie** [comment]
@DeeDeeG a note about tests, Version 2.1 of the our own tester is very broken. So no, these tests do not have any weight at all.

Although after some recent work on our tester we might now be in a position to update the version here to run the tests properly, will have to see if that could work as I forget exactly where progress on that last left off.

**Comment by DeeDeeG** [comment]
@pulsar-edit/core any thoughts on this?

- Basically it's whether we want this package to be hackable and work in its `master` branch state, such as by using `ppm link` (by merging this PR)
- or not merge this PR and have one less thing to change if we transpile out to `dist/` dir and tag a release, but the state of `master` branch requires manually editing `package.json` to be functional in the mean-time.

**Comment by Spiker985** [comment]
It looks like the github version we're using (which is atom's still as of 1.101.2023.01.19003 x64), has an `atomTranspilers` key in the package.json - which is what appears to be used by `@atom/babel7-transpiler` to do an on-the-fly transpilation in conjunction with the `.babelrc` file to tell babel what to do with it

Our version _also_ has this key, and _should_ work as intended without problems - unless a file was removed by accident

**Comment by DeeDeeG** [comment]
@Spiker985 transpilation on-demand (live when loading the package in the editor) works, at least it does when loading the package located/symlinked at `~/.pulsar/packages/github`, but it doesn't work when the `github` package is bundled/ a core package in the editor, for some reason. Either because the files aren't in the same place, or due to some of it being in `app.asar`, or just the app skipping transpiling core/bundled packages...?

So until someone figures out how to transpile a bundled/core package at run-time/on first run, we still have to transpile before bundling.

---

Strictly looking at this PR's change: We can write the transpiled files right back out to the `lib/` dir, instead of to a separate `dist/` dir. Then we wouldn't need to mess with the `"main"` field of package.json again! Regardless of whether we _do_ or _don't_ need to pre-transpile.

So I'd like to merge this if possible, to make hacking on this package easier.

**Comment by confused-Techie** [comment]
> Honestly, doing the transpilation ahead of time, and committing it is probably for the best. Since we want to get away from "magic" processes

Gotta say I agree with the sentiment here, can't remember all the details of this PR, but I'm on board with keeping build tool outputs in the actual repo to keep our life, and Pulsar build times simple

**Comment by DeeDeeG** [comment]
This PR is, make the package work when `ppm link`ed.

And implicitly the thing to do after this PR  is to transpile out to `lib/` instead of to `dist/`.

Like so:

- Before: `ELECTRON_VERSION=12 babel lib -d dist --compact auto --source-type module` (Note: `-d dist`)
- After: `ELECTRON_VERSION=12 babel lib -d lib --compact auto --source-type module` (Note: `-d lib`)

And sure, we could look into transpiling once and just committing from there, but babel's output isn't to pretty at the moment, IMO. So I would hope we could output something nicer if we're going to commit that once and for all.

I also assume that JSX transpiles out to something weird, but eh, that's an assumption.

**Comment by DeeDeeG** [comment]
Can I get an approve on this? It's a one line change.

**What's the benefit?**: this PR fixes a thing (making the project `ppm link`, and probably makes it installable directly from its GitHub repo URL, for that matter, without it being broken in that use-case).

**What's the cost?**: in exchange for the benefit above, this requires a _very small_ behavior change (-4,+3 chars) from whoever knew the magical incantation to transpile this repo before, which I think was only ever Mauricio. The behavior change in question?: Use the `-d lib` flag with babel, not `-d dist`. (As fully spelled out in my comment above).

THANK YOU IN ADVANCE IF I GET AN APPROVE.

**Comment by DeeDeeG** [comment]
It's been almost three weeks, I am absolutely going to merge this -4/+3 chars commit that I view as a strict correctness improvement, if I don't get an approve or disapprove in the next week. Thank you for your understanding.

A reminder, or a clarification, (or a whatever, I don't even know anymore at this point): A lot of this conversation is out of scope for fixing this package for the `ppm link` and `ppm install https://github.com/pulsar-edit/github` workflows.

I won't improve the transpilation workflow overall in this PR, it is way, way _**way**_ out of scope. This PR _just_ means we will start transpiling out to `lib/`, not `dist/`, and if that doesn't mean anything to you as a reviewer, or if it's over your head, please don't worry about it -- since I know how to run the babel command, and so does Mauricio, and nobody else has attempted it that I heard of, so the only two people who could be affected (me + mauricio) will decidedly be fine. It is again a -4/+3 change to the babel command required after this PR.

If that sounds too good or too simple to be true, well it's not. This really is a tiny drive-by correctness fix. If it seems too simple, it's because _it **is**_ too simple for the conversation that has been going on here. It ain't as complicated as all that. That's stuff we _could hypothetically_ do, not stuff I want to do in this PR.

THANK YOU!!!

/losing my mind a bit

**Review by confused-Techie — APPROVED** [review]
Sorry for not doing this earlier, lets merge!

**Comment by confused-Techie** [comment]
@DeeDeeG Seems I never fully reviewed the changes here, and got caught up on the content of the conversation around it. So sorry for dwelling on our of scope factors.

But this one in my opinion is good for a merge, so we can go ahead

**Comment by DeeDeeG** [comment]
Lol, disregard my rant unless anyone needed a clarification as to how this conversation diverged from my fairly narrow intend with the actual commit diff, the actual change in the PR.

**Pull request merged**
